crates/dprint/src/paths.rs

```diff
@@ -128,19 +128,15 @@ fn get_plugin_patterns<'a>(plugins: impl Iterator<Item = &'a PluginWithConfig>)
   let mut file_exts = HashSet::new();
   let mut association_globs = Vec::new();
   for plugin in plugins {
-    let mut had_positive_association = false;
     if let Some(associations) = plugin.associations.as_ref() {
       for pattern in process_config_patterns(associations) {
         if !is_negated_glob(&pattern) {
-          had_positive_association = true;
           association_globs.push(pattern);
         }
       }
     }
-    if !had_positive_association {
-      file_names.extend(&plugin.file_matching.file_names);
-      file_exts.extend(&plugin.file_matching.file_extensions);
-    }
+    file_names.extend(&plugin.file_matching.file_names);
+    file_exts.extend(&plugin.file_matching.file_extensions);
```

Comment:
What extension do you need for the prettier plugin that it currently doesn't support? I'm not sure about this change because it will certainly break someone.

Reply:
Tencent's miniprogram has a `.wxml` extension file, which is similar to HTML. In Prettier, i can configure its parser through overrides.  https://github.com/prettier/prettier/pull/9081#issuecomment-681927225

This change could indeed have unexpected impacts on existing configurations. A better approach would be to modify the Prettier plugin to support returning the configured `*.parser` association.

Currently, I support it by manually configuring all the files that need to be formatted using Prettier, so I will close this.